Agave v3.1 harness adjustments

cmoyes-jump opened the pull request.

mjain-jump commented on src/block.rs:
```diff
@@ -543,23 +549,25 @@ pub fn execute_block(context: BlockContext) -> Option<BlockEffects> {
-        None,
-        false,
-        0,
+        accounts_data_size_initial, // precomputed above
```
LOL

mjain-jump commented on src/block.rs:
```diff
@@ -543,23 +549,25 @@ pub fn execute_block(context: BlockContext) -> Option<BlockEffects> {
+    // Register the blockhashes in the bank using the new interface.
+    for blockhash in context.blockhash_queue.iter() {
+        let blockhash_hash = Hash::new_from_array(blockhash.clone().try_into().unwrap());
+        bank.register_recent_blockhash_for_test(&blockhash_hash, Some(lamports_per_signature));
```
doesnt this get recovered from the snapshot (the bank fields)?

mjain-jump commented on src/block.rs:
```diff
@@ -580,14 +588,13 @@ pub fn execute_block(context: BlockContext) -> Option<BlockEffects> {
 
-    bank.get_transaction_processor().reset_sysvar_cache();
+    // NOTE: reset_sysvar_cache and fill_missing_sysvar_cache_entries no longer
+    // needed in Agave v3.1
```
i really hope this doesnt give us any problems...

mjain-jump commented on src/lib.rs:
```diff
@@ -806,6 +847,21 @@ fn create_invoke_context_fields(
+        .collect::<Vec<_>>()
+        .into();
+    let initial_cu_avail = input.cu_avail;
+    let _accounts_snapshot = input.accounts.clone();
```
remove underscore if unused or remove entirely?

cmoyes-jump replied:
seems like this is actually unused, removing

mjain-jump commented on src/txn_fuzzer.rs:
```diff
@@ -412,38 +412,23 @@ pub fn execute_transaction(context: &TxnContext) -> Option<TxnResult> {
+    // per-signature cost when staging recent blockhashes.
+    for blockhash in &context.blockhash_queue {
+        let blockhash_hash = Hash::new_from_array(blockhash.clone().try_into().unwrap());
+        bank.register_recent_blockhash_for_test(&blockhash_hash, None);
```
i dont think this will work - you will have to register the blockhash with the correct lamports per signature otherwise the fee calculations will be incorrect / recent blockhash sysvar account state will mismatch between agave and FD (as of about a year ago, but our sysvar cache has undergone changes since then so it might not apply but worth checking out)

mjain-jump replied:
but either way, you should probably be registering the lamports per signature value

mjain-jump commented on src/vm_interp.rs:
```diff
@@ -209,28 +242,22 @@ pub fn execute_vm_interp(syscall_context: SyscallContext) -> Option<SyscallEffec
         .get_current_instruction_context()
         .unwrap();
-    let (_aligned_memory, input_memory_regions, acc_metadatas) = serialize_parameters(
+    let (_aligned_memory, input_memory_regions, acc_metadatas, _) = serialize_parameters(
```
fun fact - i got a mismatch once when using `_` instead of `_aligned_memory` because that somehow causes some borrow lifetimes to end prematurely (since theres some raw pointer accessing / storing done for the input data regions). be really careful with using `_`!!

cmoyes-jump replied:
that is crazy, the fourth parameter is a `usize`, so it should be okay, but out of paranoia, I'm adding a placeholder

mjain-jump commented on src/vm_syscalls.rs:
```diff
@@ -156,18 +201,20 @@ pub fn execute_vm_syscall(input: SyscallContext) -> Option<SyscallEffects> {
     match invoke_ctx.push() {
         Ok(_) => (),
-        Err(_) => {
+        Err(_err) => {
```
this can stay `_`

cmoyes-jump replied:
resolved

mjain-jump commented on src/vm_syscalls.rs:
```diff
@@ -186,7 +233,7 @@ pub fn execute_vm_syscall(input: SyscallContext) -> Option<SyscallEffects> {
     //   4. input data aka accounts
     // The stack gap size is 0 iff direct mapping is enabled.
-    let (_aligned_memory, input_memory_regions, acc_metadatas) = serialize_parameters(
+    let (_aligned_memory, input_memory_regions, acc_metadatas, _) = serialize_parameters(
```
see other comment

mjain-jump reviewed (changes requested):
see comments

mjain-jump committed:
ci: make persistent test vectors

cmoyes-jump committed:
Pull in latest agave-v3.0.3 commits into agave-v3.1.0-beta.0 (#396)

* bump protosol crate version in gen cargo script (#390)

* README: Include note on local Agave setup (#391)

* elf: move to flatbuffers (#392)

* elf: fix conversion script (#394)

---------

Co-authored-by: ravyu-jump <146378768+ravyu-jump@users.noreply.github.com>
Co-authored-by: mjain-jump <150074777+mjain-jump@users.noreply.github.com>

README: Include note on local Agave setup (#391)

cmoyes-jump committed:
# This is a combination of 4 commits.
# This is the 1st commit message:

Pull in latest agave-v3.0.3 commits into agave-v3.1.0-beta.0 (#396)

* bump protosol crate version in gen cargo script (#390)

* README: Include note on local Agave setup (#391)

* elf: move to flatbuffers (#392)

* elf: fix conversion script (#394)

---------

Co-authored-by: ravyu-jump <146378768+ravyu-jump@users.noreply.github.com>
Co-authored-by: mjain-jump <150074777+mjain-jump@users.noreply.github.com>
# This is the commit message #2:

README: Include note on local Agave setup (#391)


# This is the commit message #3:

elf: move to flatbuffers (#392)


# This is the commit message #4:

elf: fix conversion script (#394)

cmoyes-jump committed:
Pull in latest agave-v3.0.3 commits into agave-v3.1.0-beta.0 (#396)

* bump protosol crate version in gen cargo script (#390)

* README: Include note on local Agave setup (#391)

* elf: move to flatbuffers (#392)

* elf: fix conversion script (#394)

---------

Co-authored-by: ravyu-jump <146378768+ravyu-jump@users.noreply.github.com>
Co-authored-by: mjain-jump <150074777+mjain-jump@users.noreply.github.com>

README: Include note on local Agave setup (#391)

elf: move to flatbuffers (#392)

elf: fix conversion script (#394)

Agave v3.1 bump

PR comments round one

Add extra variable hint

PR feedback

adjust cargo toml to pull in dev feature

add back tracing

syscalls: fix mapping with custom memory module (#393)

Agave v3.1 bump

PR comments round one

Add extra variable hint

PR feedback

adjust cargo toml to pull in dev feature

add back tracing

syscalls: fix mapping with custom memory module (#393)

bump protosol crate version in gen cargo script (#390)

README: Include note on local Agave setup (#391)

Pull in latest agave-v3.0.3 commits into agave-v3.1.0-beta.0 (#396)

* bump protosol crate version in gen cargo script (#390)

* README: Include note on local Agave setup (#391)

* elf: move to flatbuffers (#392)

* elf: fix conversion script (#394)

---------

Co-authored-by: ravyu-jump <146378768+ravyu-jump@users.noreply.github.com>
Co-authored-by: mjain-jump <150074777+mjain-jump@users.noreply.github.com>

README: Include note on local Agave setup (#391)

elf: move to flatbuffers (#392)

elf: fix conversion script (#394)

Agave v3.1 bump

PR comments round one

Add extra variable hint

PR feedback

adjust cargo toml to pull in dev feature

add back tracing

syscalls: fix mapping with custom memory module (#393)

Agave v3.1 bump

PR comments round one

Add extra variable hint

PR feedback

adjust cargo toml to pull in dev feature

add back tracing

syscalls: fix mapping with custom memory module (#393)

bump protosol crate version in gen cargo script (#390)

README: Include note on local Agave setup (#391)

cmoyes-jump committed:
Update SHA

mjain-jump reviewed (commented):
dont forget to update the hardcoded and supported features lists in lib.rs

mjain-jump commented on src/vm_syscalls.rs:
```diff
@@ -59,23 +62,46 @@ fn cleanup_static_ptrs(
             Box::from_raw(program_cache_for_tx_batch_ptr as *mut ProgramCacheForTxBatch);
-        let _instr_ctx_droppable = Box::from_raw(instr_ctx_ptr as *mut InstrContext);
         let _runtime_features_droppable = Box::from_raw(runtime_features_ptr as *mut SVMFeatureSet);
+        let _instr_ctx_droppable = Box::from_raw(instr_ctx_ptr as *mut InstrContext);
+        let _callback_context_droppable =
+            Box::from_raw(callback_context_ptr as *mut SnapshotInvokeContext);
+        let _environments_droppable = Box::from_raw(
+            environments_ptr
+                as *mut solana_program_runtime::loaded_programs::ProgramRuntimeEnvironments,
+        );
```
unrelated to your PR but it would be nice to remove these static variables in the future. dont know a simple way around it. no need to address it in this pr but just something to keep in mind

mjain-jump reviewed (commented):
make sure to update the ci yml files as well to point to the correct branch

cmoyes-jump committed:
Update CI
